native/machine-coding/quiz-app-with-timer/index.js

```diff
@@ -9,52 +9,52 @@ const restart = document.getElementById("restart");
 const userScore = document.getElementById("user-score");
 const startScreen = document.querySelector(".start-screen");
 const startButton = document.getElementById("start-button");
-let questionCount = 0; // Initialize questionCount
+let questionCount = 0;
 let scoreCount = 0;
 let count = 11;
 let countdown;
+let selectedQuestions; // Declare selectedQuestions variable
 
 // Quiz questions array
 let quizArray;
 
-// Fetch the questions from the JSON file
+// Function to shuffle and select random questions
+function selectRandomQuestions(questions, count) {
+  const shuffledQuestions = questions.slice().sort(() => Math.random() - 0.5);
+  return shuffledQuestions.slice(0, count);
+}
+
+// Fetch the questions from the JSON file and initialize the quiz
 fetch("questions.json")
-    .then((response) => response.json())
-    .then((data) => {
-        quizArray = data;
-        quizCreator();
-        initial(); // Initialize the quiz
-    })
-    .catch((error) => {
-        console.error("Error loading questions:", error);
-    });
+  .then((response) => response.json())
+  .then((data) => {
+    quizArray = data;
+    initial(); // Initialize the quiz with random questions
+  })
+  .catch((error) => {
+    console.error("Error loading questions:", error);
+  });
 
-//Restart Quiz
+// Restart Quiz
 restart.addEventListener("click", () => {
   initial();
   displayContainer.classList.remove("hide");
   scoreContainer.classList.add("hide");
 });
 
-//Next Button
+// Next Button
 nextBtn.addEventListener(
   "click",
   (displayNext = () => {
```

Comment:
It would be more readable if you declare this `displayNext` function separately and then use it inside this event listener.

Reply:
okay i will do that